Source/WebCore/animation/KeyframeEffect.cpp

```diff
@@ -1316,7 +1316,11 @@ void KeyframeEffect::apply(RenderStyle& targetStyle, const Style::ResolutionCont
 
     auto computedTiming = getComputedTiming(startTime);
     if (!startTime) {
-        m_phaseAtLastApplication = computedTiming.phase;
+        if (m_phaseAtLastApplication != computedTiming.phase) {
+            m_phaseAtLastApplication = computedTiming.phase;
+            m_target->invalidateStyleAndLayerComposition();
+        }
```

Comment:
I'm not sure this is the right time to call this. You'll note that we otherwise call `invalidateStyleForAnimation()` at moments when timing properties change, but the `apply()` method is already likely called after the target was marked for invalidation and style resolution is in progress. Cc'ing @anttijk who might have some thoughts on the topic.

Reply:
That's a good point.

`invalidateStyleForAnimation` doesn't actually work here, since the transition from 'active' to 'after' might not actually have a style difference (and frequently seems to not with view transitions).

`invalidateStyleForAnimation` for the resulting restyle to have `StyleDifference::RecompositeLayer`, even if the computed style is otherwise identical.

`RenderLayerCompositor` makes different layerization decisions based on phase (using `isCurrentlyAffectingProperty`), so it also needs to know when the result of this will change.

I guess we ned to either to either find the right spot to replace a `invalidateStyleForAnimation` call with `invalidateStyleAndLayerComposition`, or alternatively just invalidate the layer composition state more directly (which feels like a bit of a layering violation here).

Reply:
Though about this more, and I don't think we can compute it earlier, since it depends on the `startTime` passed to apply.

The caller of `apply` currently computes an `AnimationImpact`, so I've refactored it a bit so that we can include this required composition into that.